File: UVA/Mathematics/BigInteger/10013.cpp
/*
Lo hice en java con BigInteger y me dio TLE

En c++

sumar dos numeros de hasta m =  un millon de digitos

crear dos vectores de tamaño m 
definir la suma para dichos vectores
	Tomando en cuenta el acarreo
	si la suma da mas de 9 se cambia el acarreo a 1 de  lo contrario cero
	para saber el valor  de la suma en ese digito solo es  mod 10

No olvidar que  debe haber un linea de impression entre cada salida 



CASO ESPECIAL  la suma se puede desbordar por 
eso se debe de tomar en cuenta el ultimo carr
porejemplo 
9
9
18 ese 1 esta en l ultimo carry pues los vectores son de tamaño 
*/


#include <bits/stdc++.h>

using namespace std;

typedef vector<int> vi;

int main(){
	int cases;
	cin >> cases;
	bool flag = true;
	while(cases--){
		if(flag)flag = false;
		else cout << endl;
		int m; cin >> m;
		vi a(m), b(m), c(m);
		for(int i = 0; i < m ; i++) cin >> a[i] >> b[i];
		int dig = 0, carry = 0; 
		for(int i = m - 1 ; i >= 0 ; i--){
			dig =  a[i] + b[i] + carry;
			if(dig > 9 ) carry = 1;
			else carry = 0;
			c[i] = dig % 10;
		}
		if(carry) printf("1");
		for(int i = 0; i < m ; i++) printf("%d", c[i]);
		printf("\n");
	}
}

File: UVA/Mathematics/BigInteger/10013.java
/*
Usar biginteger de esta manera me da tle
Opto por hacerlo en c++ con arrays

*/


import java.util.*;
import java.math.*;

class Main{
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int cases = sc.nextInt();
		boolean flag = true;
		while(cases-- > 0 ){
			if(flag)flag = false;
			else System.out.println();
			String n1 = "", n2 = "";
			int m = sc.nextInt();
			int d1,d2;
			for(int i = 0; i < m; i++){
				d1 = sc.nextInt(); n1 += d1;
				d2 = sc.nextInt(); n2 += d2;
			}
			BigInteger b1,b2;
			b1 = new BigInteger(n1);
			b2 = new BigInteger(n2);
		
			System.out.println(b1.add(b2));
		}
	}
}